### PR TITLE
feat: dog spec

### PR DESCRIPTION
**Pull request by gusfune**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

This adds specs for `dog` as requested on issue #702

### CODE REVIEW

File: src/dog.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
+const args: Fig.SingleOrArray<Fig.Arg> = {
+  name: "arguments",
+};
+
+const options: Fig.Option[] = [
+  {
+    name: ["-q", "--query"],
+    insertValue: "{HOST}",
```
Comment (QuiiBz) [inline src/dog.ts:8]: Not sure what you wanted to achieve with `insertValue` here?

File: src/dog.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  {
+    name: "--time",
+    description: "Print how long the response took to arrive",
+  },
+];
+
+const completionSpec: Fig.Spec = {
+  name: "dog",
+  description: "A command-line DNS client",
+  args: {},
```
Comment (QuiiBz) [inline src/dog.ts:84]: You can safely remove this if `dog` doesn't take any argument at the root-level

**Review by QuiiBz — CHANGES_REQUESTED** [review]

File: src/dog.ts (hunk 2 above)
Comment (gusfune) [reply src/dog.ts:84]: Dog can take arguments or no arguments at root-level. That's why I left this option. Shall I remove it?

Comment (QuiiBz) [reply src/dog.ts:83]: In this case, you should add information to each possible argument, such as the name, description, if it's variadic or not etc

Comment (QuiiBz) [reply src/dog.ts:83]: Still waiting for this, and then we'll be ready to merge!

Comment (gusfune) [reply src/dog.ts:83]: Sorry for the delay.
Working this out in a commit soon.

**Review by QuiiBz — APPROVED** [review]
Thanks for fixing the issues! LGTM 🚀

**Pull request merged**